lib/components/viewers/next-arrival-for-pattern.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react'
+
+import { stopTimeComparator } from '../../util/viewer'
+
+import StopTimeCell from './stop-time-cell'
+
+/**
+ * Shows the next arrival for a pattern within the related stops view.
+ */
+function NextArrivalForPattern (props) {
+  const {
+    homeTimezone,
+    pattern,
+    route,
+    stopTimes,
+    stopViewerArriving,
+    stopViewerConfig,
+    timeFormat
+  } = props
+
+  // sort stop times by next departure
+  let sortedStopTimes = []
+  const hasStopTimes = stopTimes && stopTimes.length > 0
+  if (hasStopTimes) {
+    sortedStopTimes = stopTimes
+      .concat()
+      .sort(stopTimeComparator)
+      // We request only x departures per pattern, but the patterns are merged
+      // according to shared headsigns, so we need to slice the stop times
+      // here as well to ensure only x times are shown per route/headsign combo.
+      // This is applied after the sort, so we're keeping the soonest departures.
+      .slice(0, stopViewerConfig.numberOfDepartures)
+  } else {
+    // Do not render pattern row if it has no stop times.
+    return null
+  }
+
+  const routeName = route.shortName ? route.shortName : route.longName
+  const title = `${routeName} To ${pattern.headsign}`
```

Comment:
`pattern.headsign` sometimes includes `routeName`. However, not on all feeds. It might be worth creating a new method that takes in a routeName and pattern headsign and splits it if the route is duplicated? 

<img width="296" alt="Screen Shot 2021-09-14 at 11 31 14 AM" src="https://user-images.githubusercontent.com/86619099/133242154-e15309eb-bca8-423f-8509-37b6987a379a.png">